### PR TITLE
[Build] Rework wx/CMakeLists.txt

### PR DESCRIPTION
**Pull request by Steelskin**
* Bring the wx frontend more in-line with the rest of the codebase.
* Always default to Unicode APIs on Windows.
* Clean up all include guards and headers.

### CODE REVIEW

**Comment by Steelskin** [comment]
I had a really heavy hand with that one. Please let me know if I removed too much. It's pretty much a full rewrite of the file.

File: src/wx/CMakeLists.txt [hunk 1]
```diff
@@ -38,488 +168,235 @@ if(NOT WIN32 AND NOT APPLE)
...
+if(NOT DEFINED WX_HAS_OPENGL)
+    check_cxx_source_compiles("
 #include <wx/wx.h>
 #include <wx/config.h>
 #include <wx/glcanvas.h>
 
 int main(int argc, char** argv) {
-    wxGLCanvas canvas(NULL, wxID_ANY, NULL, wxPoint(0, 0), wxSize(300, 300), 0);
-    return 0;
+wxGLCanvas canvas(NULL, wxID_ANY, NULL, wxPoint(0, 0), wxSize(300, 300), 0);
```
Comment (rkitover) [inline src/wx/CMakeLists.txt:316]: We could get rid of this too, the gl `COMPONENTS` check should be sufficient. I don't remember why I did this exactly, I think the `COMPONENTS` check wasn't working, but all of this likely works fine now.

**Review by rkitover — CHANGES_REQUESTED** [review]

File: src/wx/CMakeLists.txt (hunk 1 above)
Comment (Steelskin) [reply src/wx/CMakeLists.txt]: Thanks, done!

File: src/wx/CMakeLists.txt [hunk 2]
```diff
@@ -38,488 +168,235 @@ if(NOT WIN32 AND NOT APPLE)
...
+endif()
+target_include_directories(visualboyadvance-m PRIVATE ${OPENAL_INCLUDE_DIR})
+target_link_libraries(visualboyadvance-m ${OPENAL_LIBRARY})
 
-                if(${${WX_ABI_VAR}})
-                    set(WX_ABI_FOUND_MATCH TRUE)
-                    break()
-                endif()
+# XAudio2.
+if(ENABLE_XAUDIO2)
```
[the remark this answers is not in the document]
Comment (Steelskin) [reply src/wx/CMakeLists.txt:259]: Let's make this a follow-up task. Though, this doesn't build with clang-cl right now and we may just wait until FAudio is ready to ship to remove it altogether.

File: CMakeLists.txt [hunk 3]
```diff
@@ -366,24 +336,10 @@ if(NOT TRANSLATIONS_ONLY)
...
 endif()
-install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/vba-over.ini DESTINATION ${CMAKE_INSTALL_FULL_DATADIR}/vbam)
 
-# manual pages
-if(UNIX)
-    if(ENABLE_WX)
-        install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/debian/visualboyadvance-m.6 DESTINATION ${CMAKE_INSTALL_FULL_MANDIR}/man6)
-    endif()
-endif()
+add_subdirectory(po/wxvbam)
```
[the remark this answers is not in the document]
Comment (Steelskin) [reply CMakeLists.txt:342]: Yeah, I thought about it but didn't want to move the files, this PR is big enough as it is and I pray I haven't broken the release bots with anything.

**Review by rkitover — APPROVED** [review]

**Pull request merged**